[BUG] rapids-cmake is not pinned for build-sparkrapidsjni

**Describe the bug**
rapids-cmake is pinned for build-libcudf but not for build-libcudfjni, and not for build-sparkrapidsjni executions in pom.xml (h/t @robertmaynard) 

build-sparkrapidsjni execution fails at 88c03c98b1f44aac9f3e3ff7d86d752661c51cb6 with a flood of errors to the tune of 

```
/opt/rh/gcc-toolset-11/root/usr/bin/ld: libcudf.so: undefined reference to `cudf::detail::sequence
```

because libcudf ends up being built against CCCL 2.2.0
whereas [](url)

**Steps/Code to reproduce bug**

```bash
$ git checkout -f 88c03c98b1f44aac9f3e3ff7d86d752661c51cb6
$ git submodule update --init --recursive --force
$ ./build/build-in-docker clean install \
  -DGPU_ARCHS=NATIVE \
  -DBUILD_TESTS=ON \
  -DskipTests -Dlibcudf.clean.skip=false
```

--- a/pom.xml
+++ b/pom.xml
@@ -429,6 +429,8 @@
                   <arg value="-DCUDF_USE_PER_THREAD_DEFAULT_STREAM=${CUDF_USE_PER_THREAD_DEFAULT_STREAM}"/>
                   <arg value="-DRMM_LOGGING_LEVEL=${RMM_LOGGING_LEVEL}"/>
                   <arg value="-DUSE_GDS=${USE_GDS}"/>
+                  <arg value="-DCUDF_DEPENDENCY_PIN_MODE=${libcudf.dependency.mode}"/>
+                  <arg value="-C=${cudf.pin.path}/setup.cmake"/>
                 </exec>
                 <exec dir="${libcudfjni.build.path}"
                       failonerror="true"
@@ -461,6 +463,8 @@
                   <arg value="-DBUILD_BENCHMARKS=${BUILD_BENCHMARKS}"/>
                   <arg value="-DBUILD_FAULTINJ=${BUILD_FAULTINJ}"/>
                   <arg value="-DBUILD_PROFILER=${BUILD_PROFILER}"/>
+                  <arg value="-DCUDF_DEPENDENCY_PIN_MODE=${libcudf.dependency.mode}"/>
+                  <arg value="-C=${cudf.pin.path}/setup.cmake"/>
                 </exec>
                 <exec dir="${native.build.path}"
                       failonerror="true"

